### DISCUSSION TITLE
Array format

### DISCUSSION THREAD

**Discussion by cdedreuille** [supabase/supabase#803] [Questions]
Hi, 👋 I'm getting a bit confused on what's the best way to store arrays in Supabase.

I created a table like this:

<img width="328" alt="Screenshot 2021-03-04 at 12 41 46" src="https://user-images.githubusercontent.com/1540635/109958881-051cdf80-7ce7-11eb-84da-666af1ce5729.png">

I store the array like this `["adult"]`. And when I query the data I actually get an array so that's fine.

<img width="360" alt="Screenshot 2021-03-04 at 12 42 20" src="https://user-images.githubusercontent.com/1540635/109958943-18c84600-7ce7-11eb-8557-fc81978691c2.png">

But when I open the editor, the array is now an object and without editing anything I can't save it because it says that the format is not correct.

<img width="219" alt="Screenshot 2021-03-04 at 12 44 13" src="https://user-images.githubusercontent.com/1540635/109959111-5a58f100-7ce7-11eb-9ab7-120863f47136.png">
<img width="501" alt="Screenshot 2021-03-04 at 12 38 10" src="https://user-images.githubusercontent.com/1540635/109959122-5dec7800-7ce7-11eb-9c7f-5406c472e121.png">

What am I missing here?

**Comment by steve-chavez** [comment]
> But when I open the editor, the array is now an object

Can't seem to reproduce this. How do you get to the editor here? Right click, edit button in the row?

**Reply by cdedreuille** [reply]
Yes, right-click click. Here is a quick video to show the problem.

https://user-images.githubusercontent.com/1540635/110003308-89d32200-7d16-11eb-9188-0f035cc85e08.mp4

**Reply by cdedreuille** [reply]
@steve-chavez Any idea why this is happening? When I'm trying to filter it doesn't seem to work either. I get this error:

<img width="412" alt="Screenshot 2021-03-12 at 15 52 36" src="https://user-images.githubusercontent.com/1540635/110956606-169b6280-834b-11eb-95da-f3398ba2d8cf.png">

**Reply by steve-chavez** [reply]
@cdedreuille It looks like a bug.

For filtering, could you try with `{"female"}`(note the brackets) in the meantime?

(An array of many elements would be like `{"female", "male"}`, this is a postgres representation)

**Reply by cdedreuille** [reply]
Oh wow. Coming from Firebase this seems really weird. I'm not sure if I'm the only one in that case but it would be great to see this in the docs [here](https://supabase.io/docs/client/insert).

Does it mean that I need to write it like this to add an array?

const { data, error } = await supabase
  .from('cities')
  .insert([
    { name: {'first', 'second'}}
  ])

**Reply by steve-chavez** [reply]
> Does it mean that I need to write it like this to add an array?

@cdedreuille No. What I suggested above was just a workaround for the `Table Editor`(the web UI), the supabase library works normally with json arrays.

**Reply by cdedreuille** [reply]
Ah yes sorry. I tried filtering with `{"female"}` and it worked. Out of clarity, what's the best way to save an array then? Is `.insert([{ name: ['first', 'second']}])` ok?

**Reply by steve-chavez** [reply]
>  Out of clarity, what's the best way to save an array then? Is .insert([{ name: ['first', 'second']}]) ok?

Yes, that's good. Any json array will work with the library.
